sourcegraph/sourcegraph
# Code insights: Add dashboard creation UI

Closes https://github.com/sourcegraph/sourcegraph/issues/22228

### Background

Since code insight is going to have a dashboard system for sharing and viewing insights in a more convenient way we have to provide some UI/creation UI that a user be able to create their own dashboard. 
This PR adds a simple form/creation UI for the new dashboard entity. 

<img width="739" alt="Screenshot 2021-07-05 at 14 35 40" src="https://user-images.githubusercontent.com/18492575/124465746-70161a00-dd9e-11eb-9278-6e4d00fa7d35.png">

### How to test
1. Go to `/insights/add-dashboard`
2. Fill out the form and click create a dashboard
3. You should be redirected to the newly created dashboard page


### What have been done

- [x] Create dashboard area routes
- [x] Craft creation dashboard page layout
- [x] Add setting change-logic
- [x] Changing file structure according to new dashboard routes
